lib/beacon/media_library.ex

```diff
@@ -347,4 +347,51 @@ defmodule Beacon.MediaLibrary do
       _ -> :error
     end
   end
+
+  @doc """
+  Returns the contents of an uploaded file as a binary.
+  """
+  @spec read_binary(UploadMetadata.t()) :: {:ok, binary()} | {:error, atom()}
+  def read_binary(metadata) do
+    cond do
+      metadata.node == Node.self() ->
+        File.read(metadata.path)
+
+      Node.connect(metadata.node) == true ->
+        :erpc.call(metadata.node, File, :read, [metadata.path])
+
+      :otherwise ->
+        {:error, :node_not_available}
+    end
```

Comment:
Turns out this call is already [optimized by Erlang](https://github.com/erlang/otp/blob/ac7be0d00d9072e28ee9c14d1845f96dc7ed2b25/lib/kernel/src/erpc.erl#L276) so we don't need to check `Node.self()`. Besides that many other errors might raise here (including connection errors) so we'd need to capture and safely return:

```suggestion
    :erpc.call(metadata.node, File, :read, [metadata.path])
  rescue
    error -> {:error, error}
```